terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.ts

```diff
@@ -225,6 +239,46 @@ export async function scaleDown(): Promise<void> {
   }
 }
 
+export async function cleanupOldSSMParameters(runnersRgions: Set<string>, metrics: ScaleDownMetrics): Promise<void> {
+  try {
+    for (const awsRegion of runnersRgions) {
+      const ssmParams = sortSSMParaametersByUpdateTime(
+        Array.from((await listSSMParameters(metrics, awsRegion)).values()),
+      );
+
+      let deleted = 0;
+      for (const ssmParam of ssmParams) {
+        /* istanbul ignore next */
+        if (ssmParam.Name === undefined) {
+          continue;
+        }
+        if (ssmParam.LastModifiedDate === undefined) {
```

Comment:
same q as earlier about undefined dates.  Whatever you decide is the right thing to do here, it would be helpful to future maintainers to leave a comment explaining the reasoning

Reply:
I believe that this level of documentation is not necessary, justify each decision like why I decide to properly treat the correct way to handle a Typescript template even if it does not match AWS documentation is odd and too much IMO.